### ISSUE TITLE
Automatic broadcasting

### ISSUE THREAD

**Issue by jank324** [desy-ml/cheetah#138]
In #116, we added a vectorised way of using Cheetah. This means that instead of either one of these

```python
elem1 = Drift(length=0.3142)
elem2 = Drift(length=torch.tensor(0.3142))
```

you have to remember to set properties in the correct dimension even if you are not using multiple dimensions or only a single value for that property

```python
elemt3 = Drift(length=torch.tensor([0.3142])
```

By introducing automatic broadcasting (in a similar way to how PyTorch does it), we want to be able to use either one of the examples in the first code block while tracking, for example, a beam with 1000 different properties.

**Comment by cr-xu** [comment]
Is this now considered solved by #208?

**Comment by jank324** [comment]
> Is this now considered solved by #208?

Yes!
